docs/source/configuration/entity-caching.mdx

```diff
@@ -128,9 +128,162 @@ This entry contains an object with the `all` field to affect all subgraph reques
     }
 }
+```
+
+### Entity cache invalidation
+
+When existing cache entries need to be replaced, the router supports a couple of ways for you to invalidate entity cache entries:
+- **Invalidation endpoint** - the router exposes an invalidation endpoint that can receive invalidation requests from any authorized service out-of-band. A subgraph can use this endpoint to asynchronously trigger invalidation events, or a platform operator can use it to maintain the caching levels.
+- **Subgraph response extensions** - you can send invalidation requests via subgraph response extensions, allowing a subgraph to invalidate cached data right after a mutation.
+
+One invalidation request can invalidate multiple cached entries at once. It can invalidate:
+- All cached entries for a specific subgraph
+- All cached entries for a specific type in a specific subgraph
+- All cached entries for a specific entity in a specific subgraph
+
+To process an invalidation request, the router first sends a `SCAN` command to Redis to find all the keys that match the invalidation request. After iterating over the scan cursor, the router sends a `DEL` command to Redis to remove the matching keys.
+
+#### Configuration
+
+You can configure entity cache invalidation both globally (`preview_entity_cache.subgraph.all.invalidation`) and per subgraph (`preview_entity_cache.subgraph.subgraphs.invalidation`). 
+
+```yaml title="router.yaml"
+preview_entity_cache:
+  enabled: true
+  redis:
+    urls: ["redis://..."]
 
+  subgraph:
+    all:
+      enabled: true
+      # global invalidation configuration
+      invalidation:
+        # address of the invalidation endpoint
+        # this should only be exposed to internal networks
+        listen: "127.0.0.1:3000"
+        path: "/invalidation"
+        # base64 string that will be provided in the `Authorization: Basic` header value
+        shared_key: "agm3ipv7egb78dmxzv0gr5q0t5l6qs37"
+    subgraphs:
+      products:
+        # per subgraph invalidation configuration
+        invalidation:
+          # whether invalidation is enabled for this subgraph
+          enabled: true
+          # override the shared key for this particular subgraph. If another key is provided, the invalidation requests for this subgraph's entities will not be executed
+          shared_key: "czn5qvjylm231m90hu00hgsuayhyhgjv"
 ```
 
+#### Invalidation request format
+
+Invalidation requests are sent as JSON objects with the following format:
+
+- Subgraph invalidation request:
+
+```json
+{
+  "kind": "subgraph",
+  "subgraph": "accounts"
+}
+```
+
+- Subgraph type invalidation request:
+
+```json
+{
+  "kind": "subgraph",
+  "subgraph": "accounts",
+  "type": "User"
+}
+```
+
+- Subgraph entity invalidation request:
+
+```json
+{
+  "kind": "subgraph",
+  "subgraph": "accounts",
+  "type": "User",
+  "key": {
+    "id": "1"
+  }
+}
+```
+
+<Note>
+
+This is the same key format used by the router in the `representations` field for `_entities` queries.
```

Comment:
```suggestion
The key field is the same arguments as defined in the subgraph's `@key` directive. If a subgraph has multiple keys defined and the entity is being invalidated, it is likely you'll need to send a request for each key definition. 
```

Reply:
Most subgraph authors do not care or know about the `representations` arg here; it's an implementation detail for their subgraph that most teams have abstracted.

Reply:
good point yes